### PR TITLE
CB-9563 Mulptipart form data is used even a header named Content-Type…

### PR DESCRIPTION
**Pull request by daserge**
… is present

Adds non-multipart implementation for Windows and a corresponding test
Adds a fix for Android for uploadResult.bytesSent = 0 for small files
Fixes parameters of FILE_NOT_FOUND_ERR case in upload operation on Windows - passing source file name instead of duplicated server/destination argument

[Jira issue](https://issues.apache.org/jira/browse/CB-9563)

### CODE REVIEW

**Comment by vladimir-kotikov** [comment]
@daserge, don't forget to update plugin docs regarding this new behaviour for Windows platform

File: src/android/FileTransfer.java [hunk 1]
```diff
@@ -528,14 +528,18 @@ public void run() {
...
+
                     Log.d(LOG_TAG, "got response from server");
                     Log.d(LOG_TAG, responseString.substring(0, Math.min(256, responseString.length())));
-                    
+
                     // send request and retrieve response
                     result.setResponseCode(responseCode);
                     result.setResponse(responseString);
 
+                    if(result.getBytesSent() != totalBytes) {
```
Comment (vladimir-kotikov) [inline src/android/FileTransfer.java]: Is this code related to JIRA, this PR fixes? If no, consider submitting it via separate PR.

File: src/android/FileTransfer.java [hunk 2]
```diff
@@ -528,14 +528,18 @@ public void run() {
...
+                      // occurs for a small file
```
Comment (vladimir-kotikov) [inline src/android/FileTransfer.java]: nit: please correct indentation

File: src/android/FileTransfer.java (hunk 1 above)
Comment (daserge) [inline src/android/FileTransfer.java]: This code is related to [the added test](https://github.com/apache/cordova-plugin-file-transfer/pull/117/files#diff-2a8a5fef3397df87ab538f028a5c6b50R951).

File: src/windows/FileTransferProxy.js [hunk 3]
```diff
@@ -81,6 +81,8 @@ exec(win, fail, 'FileTransfer', 'upload',
         var uploadId = options[9];
         var httpMethod = options[10];
 
+        var isMultipart = typeof headers["Content-Type"] === 'undefined';
```
Comment (vladimir-kotikov) [inline src/windows/FileTransferProxy.js]: nit: it might be better to move this definition closer to usage

**Comment by vladimir-kotikov** [comment]
LGTM :ship: at will

File: src/windows/FileTransferProxy.js (hunk 3 above)
Comment (dblotsky) [inline src/windows/FileTransferProxy.js]: It might also be clearer to write out the `if` statement.
